### PR TITLE
CADC-8954: provide clientTransfer uws errorSummary check, check HttpUpload failures

### PR DESCRIPTION
**Pull request by hjeeves**
Errors from both these routes were not caught before.  Check for errorSummary is provided as a function so the calling program can decide how to handle errors.

### CODE REVIEW

**Review by brianmajor — CHANGES_REQUESTED** [review]
- Can you combine checkFailure and checkTransferFailure somehow?  I think the fact that the errors come from different sources should be hidden from users of this class.
- Use the error constants in NodeFault.java for checking the type of transfer error.
- You may as well update this whole class with the correct checkstyle.

**Comment by brianmajor** [comment]
I just realized that NodeFault.java is in cadc-vos-server.  Could you look at moving it to cadc-vos?

**Comment by brianmajor** [comment]
You probably also want to add a helper method in NodeFault.java to make the Fault->Exception mapping easier.

**Comment by hjeeves** [comment]
Moving NodeFault.java turned out to be more problematic than it was worth (would have required an underlying library upgrade that caused a clash with code requiring the older version.) So it's left as is, and will refer to the VOS constants added.

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java [hunk 1]
```diff
@@ -167,4 +179,47 @@ public static void checkFailure(Throwable failure)
...
+            if (StringUtil.hasLength(errorMsg)) {
+                if (errorMsg.contains("Invalid Argument")
+                    || VOS.IVOA_FAULT_INVALID_ARG.equals(errorMsg)) {
+                    log.debug("Invalid Argument found: " + errorMsg);
+                    throw new IllegalArgumentException(VOS.IVOA_FAULT_INVALID_ARG);
+                }
+
+                // not found
+                if (VOS.IVOA_FAULT_NODE_NOT_FOUND.equals(errorMsg) ||
+                    errorMsg.contains("not a data node")) {
```
Comment (brianmajor) [inline cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java:207]: Do you still need to check for the non-standard string messages, like "Invalid Argument" and "not a data node"? 

I've seen "not a data node" thrown quite often so perhaps you do.  This is part of a transfer negotiation step I believe, and is thrown when someone is trying to download a directory (container node) and not a file (data node).  So, it should probably be mapped to IllegalArgumentException instead.

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java [hunk 2]
```diff
@@ -364,6 +383,11 @@ public Node setNode(Node node)
         {
             throw new IllegalStateException("Node not found", e);
         }
+        catch (ResourceAlreadyExistsException e)
+        {
+            log.debug("failed to set node", e);
+            throw new IllegalStateException("failed to set node", e);
```
Comment (brianmajor) [inline cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java:389]: I'm sure there's a good reason why these all end up as IllegalStateExceptions?  I just don't remember...

**Review by brianmajor — CHANGES_REQUESTED** [review]

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java (hunk 1 above)
Comment (hjeeves) [reply cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java:207]: I've also seen 'not a data node' thrown in the past few weeks of testing. And items like 'invalidArgument' (no space) - there's only so much testing for non-standard that can be done.

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java (hunk 2 above)
Comment (hjeeves) [reply cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java:389]: As discussed in Slack - I don't know why, just see that it's used throughout cadc-vos and cadc-vos-server.

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java (hunk 1 above)
Comment (hjeeves) [reply cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java:207]: Changed 'node not found' to 'IllegalArgumentException'

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java (hunk 2 above)
Comment (hjeeves) [reply cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSpaceClient.java:389]: Believe the decision is to leave this as-is.

File: cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java (hunk 1 above)
Comment (brianmajor) [reply cadc-vos/src/main/java/ca/nrc/cadc/vos/client/VOSClientUtil.java:207]: node not found -> ResourceNotFoundException
not a data node -> IllegalArgumentException

**Review by brianmajor — CHANGES_REQUESTED** [review]

**Review by brianmajor — APPROVED** [review]

**Pull request merged**